infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java

```diff
@@ -91,6 +108,42 @@ private String generateNodeStartCommand(String instanceId, String rmUrlToUse, St
         return "nohup " + javaCommand + javaProperties + "  &";
     }
 
+    public String generateDefaultIaasConnectorURL(String DefaultRMHostname) {
+        if (nsConfig == null) {
+            try {
+                // If the configuration manager is not loaded, I load it with the NodeSource properties file
+                nsConfig = NSProperties.loadConfig();
+            } catch (ConfigurationException e) {
+                // If something go wring, I switch to hardcoded configuration, and leave.
+                logger.error("Exception when loading NodeSource properties", e);
+                return "http://" + DefaultRMHostname + ":8080/connector-iaas";
+            }
+        }
+        // I return the requested value while taking into account the configuration parameters
+        return nsConfig.getString(NSProperties.DEFAULT_PREFIX_CONNECTOR_IAAS_URL) + DefaultRMHostname +
+               nsConfig.getString(NSProperties.DEFAULT_SUFFIX_CONNECTOR_IAAS_URL);
+    }
+
+    public String generateDefaultDownloadCommand(String rmHostname) {
+        String suffixRmToNodeJarUrl;
+        if (nsConfig == null) {
+            // If the configuration manager is not loaded, I load it with the NodeSource properties file
+            try {
+                nsConfig = NSProperties.loadConfig();
+                suffixRmToNodeJarUrl = nsConfig.getString(NSProperties.DEFAULT_SUFFIX_RM_TO_NODEJAR_URL);
+            } catch (ConfigurationException e) {
```

Comment:
Same as the other method. I suggest to use throws instead of try/catch block. I prefer a fail fast situation when the config cannot be loaded properly.

Reply:
This statement has to be discussed.